ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/BlobsBundleV2.java

```diff
@@ -0,0 +1,135 @@
+import org.hyperledger.besu.ethereum.core.kzg.BlobsWithCommitments;
+import org.hyperledger.besu.ethereum.core.kzg.KZGCommitment;
+import org.hyperledger.besu.ethereum.core.kzg.KZGProof;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import ethereum.ckzg4844.CKZG4844JNI;
+import ethereum.ckzg4844.CellsAndProofs;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes48;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@JsonPropertyOrder({"commitments", "proofs", "blobs"})
+public class BlobsBundleV2 {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BlobsBundleV2.class);
+  private final List<String> commitments;
+
+  private final List<String> cellProfs;
+
+  private final List<String> blobs;
+
+  public BlobsBundleV2(final List<Transaction> transactions) {
+    final List<BlobsWithCommitments> blobsWithCommitments =
+        transactions.stream()
+            .map(Transaction::getBlobsWithCommitments)
+            .flatMap(Optional::stream)
+            .map(this::mapBlobWithCommitments)
+            .toList();
+
+    this.commitments =
+        blobsWithCommitments.stream()
+            .flatMap(b -> b.getKzgCommitments().stream())
+            .map(KZGCommitment::getData)
+            .map(Bytes::toString)
+            .collect(Collectors.toList());
+
+    this.cellProfs =
+        blobsWithCommitments.stream()
+            .flatMap(b -> b.getKzgProofs().stream())
+            .map(KZGProof::getData)
+            .map(Bytes::toString)
+            .collect(Collectors.toList());
+
+    this.blobs =
+        blobsWithCommitments.stream()
+            .flatMap(b -> b.getBlobs().stream())
+            .map(Blob::getData)
+            .map(Bytes::toString)
+            .collect(Collectors.toList());
+
+    LOG.info(
+        "BlobsBundleV2: totalTxs: {}, blobTxs: {}, commitments: {}, cell proofs: {}, blobs: {}",
+        transactions.size(),
+        blobsWithCommitments.size(),
+        commitments.size(),
+        cellProfs.size(),
+        blobs.size());
+  }
+
+  @JsonGetter("commitments")
+  public List<String> getCommitments() {
+    return commitments;
+  }
+
+  @JsonGetter("proofs")
+  public List<String> getCellProofs() {
+    return cellProfs;
+  }
+
+  @JsonGetter("blobs")
+  public List<String> getBlobs() {
+    return blobs;
+  }
+
+  @SuppressWarnings("UnusedVariable")
+  private BlobsWithCommitments mapBlobWithCommitments(
+      final BlobsWithCommitments blobsWithCommitments) {
+    if (blobsWithCommitments.getBlobType() == BlobType.KZG_CELL_PROOFS) {
+      return blobsWithCommitments;
+    }
+    LOG.info(
+        "BlobProofBundle {} versionId is 0. Converting to version {}",
+        blobsWithCommitments.getVersionedHashes(),
+        BlobType.KZG_CELL_PROOFS);
```

Comment:
s.a.

Reply:
I kept this as warn for now. This should happen only during the `Fusaka` transition while the pool still have old blob types.